src/containers/SubjectPage/SubjectContainer.jsx

```diff
@@ -149,71 +140,40 @@ const SubjectPage = ({
           },
         ]
       : []),
-    ...(topic ? [{ ...topic, isCurrent: currentLevel === 'Topic' }] : []),
-    ...(subTopic
-      ? [{ ...subTopic, isCurrent: currentLevel === 'Subtopic' }]
-      : []),
-    ...(subSubTopic
-      ? [{ ...subSubTopic, isCurrent: currentLevel === 'SubSubtopic' }]
+    ...(breadCrumbList.length > 0
+      ? breadCrumbList.map(crumb => ({
+          ...crumb,
+          isCurrent: currentLevel === crumb.index,
+          typename: crumb.index > 0 ? 'Subtopic' : 'Topic',
+        }))
       : []),
   ];
 
-  const setTopicBreadCrumb = topic => {
-    setCurrentLevel('Topic');
-    setTopic(
-      topic
-        ? {
-            ...topic,
-            typename: 'Topic',
-            url: '#',
-          }
-        : null,
-    );
-  };
-
-  const setSubTopicBreadCrumb = topic => {
-    setCurrentLevel('Subtopic');
-    setSubTopic(
-      topic
-        ? {
-            ...topic,
-            typename: 'Subtopic',
-            url: '#',
-          }
-        : null,
-    );
-  };
-
-  const setSubSubTopicBreadCrumb = topic => {
-    setCurrentLevel('SubSubtopic');
-    setSubSubTopic(
-      topic
-        ? {
-            ...topic,
-            typename: 'SubSubtopic',
-            url: '#',
-          }
-        : null,
-    );
+  const setBreadCrumb = topic => {
+    setCurrentLevel(topic.index);
+    setBreadCrumbList(prevState => [
+      ...prevState.filter(
+        b =>
+          b.typename === 'Subjecttype' ||
+          b.typename === 'Subject' ||
+          topics.includes(b.id),
```

Comment:
Forslag til løsning.
```suggestion
          b.id.localeCompare(topic.id) !== 0 &&
          (b.typename === 'Subjecttype' ||
          b.typename === 'Subject' ||
          topics.includes(b.id)),
```